src/plugins/formulas/__tests__/formulas.spec.js

```diff
@@ -2097,4 +2097,155 @@ describe('Formulas general', () => {
       ['A5']
     ]);
   });
+
+  describe('cooperation with validation', () => {
+    it('should validate result of formula properly (opening and closing an editor)', async() => {
+      const hot = handsontable({
+        data: [
+          ['=B1+5', 2, '=D1', 'text']
+        ],
+        formulas: {
+          engine: HyperFormula
+        },
+        type: 'numeric'
+      });
+
+      selectCell(0, 0);
+      // Opening an editor
+      keyDown('enter');
+      // Closing the editor and saving changes.
+      keyDown('enter');
+
+      await sleep(0); // Validator is asynchronous.
```

Comment:
Is all validation async or is it a workaround just for this plugin?

Reply:
All are async, and another tests in Handsontable reserved for that 100 ms. I will change it.